### ISSUE TITLE
While making html: AttributeError: 'Sphinx' object has no attribute 'info' 

### ISSUE THREAD

**Issue by SKPsanjeevi** [sphinx-doc/sphinx#6871]
I am not sure if this is a bug report but I am posting here as per the error message I received. I am trying to build the documentation for the `matplotlib` (https://matplotlib.org/3.1.1/devel/documenting_mpl.html#building-the-docs)

```
make html
```

The above command produced the following error message. I tried to look up online about solutions for following message but could not fix it.
```
Running Sphinx v2.2.1
WARNING: while setting up extension matplotlib.sphinxext.mathmpl: directive 'math' is already registered, it will be overridden
/home/sathish/anaconda3/lib/python3.7/site-packages/sphinx/util/docutils.py:286: RemovedInSphinx30Warning: function based directive support is now deprecated. Use class based directive instead.
  RemovedInSphinx30Warning)
/home/sathish/anaconda3/lib/python3.7/site-packages/sphinx/util/docutils.py:286: RemovedInSphinx30Warning: function based directive support is now deprecated. Use class based directive instead.
  RemovedInSphinx30Warning)

Exception occurred:
  File "/home/sathish/Documents/matplotlib/doc/sphinxext/github.py", line 149, in setup
    app.info('Initializing GitHub plugin')
AttributeError: 'Sphinx' object has no attribute 'info'
The full traceback has been saved in /tmp/sphinx-err-4mk5ue2_.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Makefile:30: recipe for target 'html' failed
make: *** [html] Error 2
```

Contents of `/tmp/sphinx-err-4mk5ue2_.log`

```
# Sphinx version: 2.2.1 
# Python version: 3.7.5 (CPython) 
# Docutils version: 0.15.2 release 
# Jinja2 version: 2.10.3 
# Last messages: 
 
# Loaded extensions: 
Traceback (most recent call last): 
  File "/home/sathish/anaconda3/lib/python3.7/site-packages/sphinx/cmd/build.py", line 275, in build_main 
    args.tags, args.verbosity, args.jobs, args.keep_going) 
  File "/home/sathish/anaconda3/lib/python3.7/site-packages/sphinx/application.py", line 236, in __init__ 
    self.setup_extension(extension) 
  File "/home/sathish/anaconda3/lib/python3.7/site-packages/sphinx/application.py", line 395, in setup_extension 
    self.registry.load_extension(self, extname) 
  File "/home/sathish/anaconda3/lib/python3.7/site-packages/sphinx/registry.py", line 487, in load_extension 
    metadata = setup(app) 
  File "/home/sathish/Documents/matplotlib/doc/sphinxext/github.py", line 149, in setup 
    app.info('Initializing GitHub plugin') 
AttributeError: 'Sphinx' object has no attribute 'info' 

```

**Comment by jfbu** [comment]
The matplotlib whose documentation you want to build is incompatible with Sphinx 2.2.1. Actually  https://github.com/matplotlib/matplotlib/commit/f87e9f46daf42ad78626e3fa5342eeacc7f19d8e#diff-811874bf3d269150ae8957797810c3e0  removed reference to `app.info()` with this message
> - sphinx 1.6 deprecated app.info in favor of a logging API, but we don't
  really need either anyways
  http://www.sphinx-doc.org/en/master/changes.html#id52

Remark that [matplotlib documentation](https://matplotlib.org/3.1.1/devel/documenting_mpl.html#installing-dependencies) for the version you gave the link to does say
```
sphinx>=1.3,!=1.5.0,!=1.6.4,!=1.7.3,!=1.8.0,<2.0.0
```
meaning no Sphinx 2.x can be used (try Sphinx v1.8.5), but the actual [matplotlib requirements](https://github.com/matplotlib/matplotlib/blob/master/requirements/doc/doc-requirements.txt) file has only
```
sphinx>=1.8.1,!=2.0.0
```
thus hopefully you can build the docs of current dev version of matplotlib with current Sphinx.

**Comment by jfbu** [comment]
Compiling (`make html`) matplotlib docs (master at matplotlib/matplotlib@3d1a1bcc) with Sphinx v2.2.1 from an editable install in a Python 3.7.5 virtualenv according to [matplotlib instructions](https://matplotlib.org/3.1.1/devel/documenting_mpl.html#installing-dependencies) completed at my locale (old macos x). But with 2 warnings:
```
WARNING: Could not find backreferences file: /path/to/matplotlib/doc/api/_as_gen/matplotlib.rcParams.examples.new
WARNING: The backreferences are likely to be erroneous due to file system case insensitivity.
```
hence `make` reported an error from the `-W` option in `SPHINXOPTS` of the `Makefile`. Apart from that, it seems things look fine for html build.

I then tried `make latexpdf SPHINXOPTS=` but it ended in a build error
```
! LaTeX Error: Too deeply nested.

See the LaTeX manual or LaTeX Companion for explanation.
Type  H <return>  for immediate help.
 ...                                              
                                                  
l.152900 \begin{itemize}
```
I started looking at the LaTeX configuration in `doc/conf.py` of matplotlib project and the above issue is encountered because `latex_elements` dictionary (which had usage of `enumitem` package which fixes the above) is overwritten. Since Sphinx 1.5, one only needs to make use Sphinx `latex_elements` `'maxlistdepth'` key, one does not need the `enumitem` package.

There are other issues in the LaTeX aspects of the conf.py that I may possibly report upstream at their project once I have satisfactory situation regarding fonts: with DejaVu Serif which this conf.py configures as font for LaTeX there are many LaTeX warnings of the type "There is no ∅ in font DejaVu" during LaTeX build (**134** distinct characters are missing from the font!) [1] It seems many Unicode math symbols are used in plain text mode. Situation is much better with GNU FreeFont which is the current default for Sphinx with `latex_engine = 'xelatex'`; only **19** missing characters but as my Terminal displays them at least I know they are available in "Menlo Regular 18pt" which is system font on my Mac... 

.. [1]:  apart from the missing `b`, `c`, `d` from  usage of `\mathbb` which comes from a section of the documentation dealing with STIX fonts, but the LaTeX is not configured to use STIX font...

**Comment by jfbu** [comment]
`Sphinx.info()` was [deprecated](http://www.sphinx-doc.org/en/master/extdev/deprecated.html) at 1.6 and removed at 2.0. The matplotlib/matplotlib@f87e9f46 commit removes its usage. This commit got merged into matplotlib 3.1.0  and [backported to v3.0.x](https://github.com/matplotlib/matplotlib/pull/12129/files) to 3.0.x series.

As reported above, your report made me look at compiling current matplotlib docs with current Sphinx, which basically looks ok for html builds and needs some fixes for latex which I submitted upstream. Thanks, now closing.
